### ISSUE TITLE
Use of `tablename:` in contenttypes disallows editing records. 

### ISSUE THREAD

**Issue by bobdenotter** [bolt/bolt#5960]
If the key and slug for a contenttype aren't the same, editing breaks. 

Details
-------

 - Relevant Bolt Version: [ 3.1 | 3.2 | master ]

Reproduction
------------

Create a contenttype, like this: 

```
blockzz:
    name: Bloxes
    singular_name: Blox
    tablename: blocks
    fields:
        title:
            type: text
            class: large
            group: "Block"
        slug:
            type: slug
            uses: [ title ]
```

Go to the dashboard, update DB, add "Loripsum dummy content". So far everything works as expected. 

Go to the `Bloxes` overview: 

![screen_shot_2016-10-26_at_15_53_11](https://cloud.githubusercontent.com/assets/1833361/19728685/a58cccd4-9b94-11e6-9522-2ab8fbe6fcd7.png)

This link _seems_ allright. At least, it's consistent with how other contenttypes generate the link. However, if you click the link, you get redirected to the frontpage, with this error: 

![screen shot 2016-10-26 at 15 56 41](https://cloud.githubusercontent.com/assets/1833361/19728739/d6af74e2-9b94-11e6-83cb-215b527a82d5.png)

Routing and matching works correctly in the frontend, though: 

![screen_shot_2016-10-26_at_15_57_57](https://cloud.githubusercontent.com/assets/1833361/19728835/2b8ab7ba-9b95-11e6-8b8f-16bfb9adc5f8.png)

Pinging @phpetra. Is this the same issue you encountered?

**Comment by bobdenotter** [comment]
Perhaps this is related to #5969. Will test too.

**Comment by bobdenotter** [comment]
The actual exception that gets thrown is: 

```
InvalidRepositoryException in EntityManager.php line 259: 
Attempted to load repository for invalid class or alias: blocks. Check that the class, alias or contenttype definition is correct.
```

The issue isn't with the discrepancy of name/slug, but rather with the use of `tablename`. This breaks in the same way: 

```

blocks:
    name: Blocks
    slug: blocks
    singular_name: Block
    singular_slug: block
    tablename: blocktable
    fields:
        title:
            type: text
            class: large
            group: "Block"
        slug:
            type: slug
            uses: [ title ]
```

@rossriley Could you take a look at this one, too?

**Comment by bobdenotter** [comment]
Perhaps noteworthy: 

 - If you use `prefill`, you can fill the contenttype just fine. 

![screen shot 2016-10-29 at 15 14 42](https://cloud.githubusercontent.com/assets/1833361/19829764/9e2bdb30-9dea-11e6-9be7-32be028e58a2.png)


 - In the _frontend_ they also work as expected. You can list them, they show up as normal. 

![screen shot 2016-10-29 at 15 16 33](https://cloud.githubusercontent.com/assets/1833361/19829770/b9a00c88-9dea-11e6-8e62-9fbb4d1d5286.png)

**Comment by bobdenotter** [comment]
Another note: This was also broken in 3.0 and 3.1, so it's not a newly introduced regression. :-)

**Comment by bobdenotter** [comment]
Fixed in #5971
